src/main/java/org/medicmobile/webapp/mobile/AppUrlVerifier.java

```diff
@@ -87,3 +91,22 @@ public static AppUrlVerification failure(String appUrl, int failure) {
 		return new AppUrlVerification(appUrl, false, failure);
 	}
 }
+
+class AppUrlVerificationTask implements Callable<AppUrlVerification> {
+	private final String appUrl;
+
+	public AppUrlVerificationTask(String appUrl) {
+		this.appUrl = appUrl;
+	}
+
+	@Override
+	public AppUrlVerification call() {
+		trace(this, "AppUrlVerificationTask :: Executing call, appUrl=%s", appUrl);
+
+		if (DEBUG && (appUrl == null || appUrl.isEmpty())) {
```

Comment:
This code is from [here](https://github.com/medic/cht-android/blob/ebe095c6432d707e5b153c1526e4b82eded2f77a/src/main/java/org/medicmobile/webapp/mobile/SettingsDialogActivity.java#L44).